### PR TITLE
#4500 Modifying application.yaml as required by openshift.io osio pipeline library

### PR DESCRIPTION
**Pull request by rupalibehera**
This changed include the below which is required by osio pipeline library (https://github.com/fabric8io/osio-pipeline) :

1. Adds parameter - RELEASE_VERSION which is basically the
build number and will be used to differentiate between
the resources of the particular build.
2. Adds parameter - SUFFIX_NAME which is an identifier to
differentiate between the master build or other branches.
More like a master build or PR build. So whenever there is
a change in the resources of a particular branch, it applies
to the resource of that branch only. The SUFFIX_NAME variable
is used in the name of all resources.

### CODE REVIEW

**Comment by geoand** [comment]
[please test]

**Comment by geoand** [comment]
@rupalibehera Thank you for this.
We'll have to look into this with @metacosm

**Comment by geoand** [comment]
test this please

**Comment by geoand** [comment]
test this please

**Comment by rupalibehera** [comment]
@geoand and @metacosm, could you guys have a look at this PR

**Comment by geoand** [comment]
Sorry @rupalibehera , this slipped through the cracks...

I have to say I don't fully understand why these changes are needed since I am missing the bigger picture.
@metacosm @Ladicek do you guys have any more insights into this?

**Comment by Ladicek** [comment]
We're having a conversation about some of the changes here: https://github.com/thorntail-examples/rest-http/pull/51 I'd suggest to wait until the point about triggering rebuilds is resolved. I disagree with what OSIO wants there, but I'm sure they have good reasons for that. So I'm kinda stuck.

**Comment by geoand** [comment]
Thanks @Ladicek! I will have a look at your link :)

**Comment by geoand** [comment]
Also see: https://github.com/openshiftio-vertx-boosters/vertx-http-booster-redhat/pull/36

**Comment by rupalibehera** [comment]
@Ladicek @geoand  could you guys take a look at this PR we fixed the suggestion mentioned by you guys

**Comment by geoand** [comment]
@rupalibehera I would like to see what will be said wrt https://github.com/openshiftio-vertx-boosters/vertx-http-booster-redhat/pull/36#issuecomment-458496449

**Comment by Ladicek** [comment]
I think it starts to make sense now. Looking at one of the Vert.x PRs, I can't object to anything in it. However, @geoand, I think that the `RUNTIME_VERSION` and `BOOSTER_VERSION` replaceables are exclusive to Spring Boot, so that's something you'll have to debate on your own.

**Comment by geoand** [comment]
> I think it starts to make sense now. Looking at one of the Vert.x PRs, I can't object to anything in it. However, @geoand, I think that the `RUNTIME_VERSION` and `BOOSTER_VERSION` replaceables are exclusive to Spring Boot, so that's something you'll have to debate on your own.

Yeah, those we have added, but they are not critical.

Thanks for your input!

**Comment by rupalibehera** [comment]
> @rupalibehera I would like to see what will be said wrt [openshiftio-vertx-boosters/vertx-http-booster-redhat#36 (comment)](https://github.com/openshiftio-vertx-boosters/vertx-http-booster-redhat/pull/36#issuecomment-458496449)

Yes we have a email thread for discussion on obsidian-team mailing list.

**Comment by geoand** [comment]
@metacosm Any objections to removing our custom placeholders?

File: .openshiftio/application.yaml [hunk 1]
```diff
@@ -12,6 +12,14 @@ metadata:
...
 parameters:
+- name: SUFFIX_NAME
+  description: The suffix name for the template objects
+  displayName: Suffix name
+  value: ''
+- name: RELEASE_VERSION
+  description: The release version number of application
+  displayName: Release version
+  value: 1.0.0
 - name: RUNTIME_VERSION
```
Comment (metacosm) [inline .openshiftio/application.yaml]: This parameter needs to be removed as well, then.

**Comment by rupalibehera** [comment]
> Have these changes been validated in Launcher? Also, that means that we will need to update our productisation process (though it should be code removal so I don't anticipate this to be an issue)…

This has been tested with OSIO which uses launcher add on, I am not sure how this should be validated in launcher. But the nodejs boosters have the same changes and they are already in master.

**Comment by rupalibehera** [comment]
@metacosm removed `RUNTIME_VERSION` parameter for all three spring-boot boosters

**Comment by rupalibehera** [comment]
@metacosm does this PR look good or if we have something else please let me know ?

**Comment by rupalibehera** [comment]
@Ladicek @kenfinnigan @metacosm could you guys take a look at this PR we fixed the suggestion mentioned by you guys

**Comment by metacosm** [comment]
Of note, using `latest` for the java image is actually causing issues as the just released `1.6` is broken… which is why it was fixed in the first place but I guess we never really put in place tests to validate new images so maybe we can assume that `latest` should work "most" of the time (despite it being currently broken).

**Comment by rupalibehera** [comment]
> Of note, using `latest` for the java image is actually causing issues as the just released `1.6` is broken… which is why it was fixed in the first place but I guess we never really put in place tests to validate new images so maybe we can assume that `latest` should work "most" of the time (despite it being currently broken).

I thought of making it consistent and while testing it it failed for not resolving the image and latest was working fine. But now I tested again with the RUNTIME_VERSION I am changing it back the way it was before in a separate PR for which have been merged.

**Pull request merged**